File: backend/src/main/java/com/shoppingmall/entity/Menu.java
package com.shoppingmall.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 菜单实体类
 *
 * @author ShoppingMall Team
 * @date 2025-12-04
 */
@Data
@TableName("sys_menu")
public class Menu {

    /**
     * 主键ID
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 父菜单ID（0表示顶级菜单）
     */
    private Long parentId;

    /**
     * 菜单名称
     */
    private String menuName;

    /**
     * 菜单类型（目录/菜单/按钮）
     */
    private String menuType;

    /**
     * 路由路径
     */
    private String path;

    /**
     * 组件路径
     */
    private String component;

    /**
     * 图标
     */
    private String icon;

    /**
     * 权限标识（如：admin:user:list）
     */
    private String permission;

    /**
     * 排序（数字越小越靠前）
     */
    private Integer sortOrder;

    /**
     * 状态（0-禁用，1-启用）
     */
    private Integer status;

    /**
     * 逻辑删除（0-未删除，1-已删除）
     */
    @TableLogic
    private Integer deleted;

    /**
     * 创建时间
     */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: admin-frontend/src/api/admin/menu.ts
/**
 * 菜单管理API
 */

import request from '@/utils/request'

// 菜单VO
export interface MenuVO {
  id?: number
  parentId?: number
  menuName: string
  menuType: string
  path?: string
  component?: string
  icon?: string
  permission?: string
  sortOrder?: number
  status: number
  children?: MenuVO[]
}

/**
 * 获取菜单树
 */
export const getMenuTree = (): Promise<MenuVO[]> => {
  return request.get('/api/admin/menu/tree')
}

/**
 * 获取菜单详情
 */
export const getMenuById = (id: number): Promise<MenuVO> => {
  return request.get(`/api/admin/menu/${id}`)
}

/**
 * 新增菜单
 */
export const addMenu = (data: MenuVO): Promise<void> => {
  return request.post('/api/admin/menu', data)
}

/**
 * 更新菜单
 */
export const updateMenu = (id: number, data: MenuVO): Promise<void> => {
  return request.put(`/api/admin/menu/${id}`, data)
}

/**
 * 删除菜单
 */
export const deleteMenu = (id: number): Promise<void> => {
  return request.delete(`/api/admin/menu/${id}`)
}

/**
 * 启用/禁用菜单
 */
export const updateMenuStatus = (id: number, status: number): Promise<void> => {
  return request.put(`/api/admin/menu/${id}/status`, null, {
    params: { status }
  })
}


